pennylane/qnode.py

```diff
@@ -197,12 +202,9 @@ def __init__(self, func, device, cache=False):
 
         self.cache = cache
 
-        self.variable_ops = {}
-        """ dict[int->list[(int, int)]]: Mapping from free parameter index to the list of
-        :class:`Operations <pennylane.operation.Operation>` (in this circuit) that depend on it.
-
-        The first element of the tuple is the index of the Operation in the program queue,
-        the second the index of the parameter within the Operation.
+        self.variable_deps = {}
+        """dict[int, list[ParDep]]: Mapping from free parameter index to the list of
+        :class:`~pennylane.operation.Operation` instances (in this circuit) that depend on it.
```

Comment:
I approve of this change :+1: The `variable_ops` dictionary was very hard to understand/parse.

Reply:
`namedtuples` are great and should be used more :)